TM4J-t használó modul-ok esetén exception

Ha egy modul-ba be van kötve a Roaster TM4J modulja, akkor az alábbi exception jön. Feltehetőleg a gondot az okozza, hogy a microprofile verzió emelve lett, de a TM4J-ben használt microprofile rest client implementáció nem lett upgrade-elve, nem kompatibilis az API-val.

```java
javax.enterprise.event.ObserverException
        at java.base/java.lang.J9VMInternals.newInstanceImpl(Native Method)
        at java.base/java.lang.Class.newInstance(Class.java:2086)
        at org.jboss.weld.security.NewInstanceAction.run(NewInstanceAction.java:33)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:734)
        at org.jboss.weld.injection.Exceptions.rethrowException(Exceptions.java:40)
        at org.jboss.weld.injection.Exceptions.rethrowException(Exceptions.java:78)
        at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:103)
        at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:85)
        at org.jboss.weld.injection.MethodInvocationStrategy$SimpleMethodInvocationStrategy.invoke(MethodInvocationStrategy.java:168)
        at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:330)
        at org.jboss.weld.event.ExtensionObserverMethodImpl.sendEvent(ExtensionObserverMethodImpl.java:123)
        at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:308)
        at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:286)
        at javax.enterprise.inject.spi.ObserverMethod.notify(ObserverMethod.java:124)
        at org.jboss.weld.util.Observers.notify(Observers.java:166)
        at org.jboss.weld.event.ObserverNotifier.notifySyncObservers(ObserverNotifier.java:285)
        at org.jboss.weld.event.ObserverNotifier.notify(ObserverNotifier.java:273)
        at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:177)
        at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:171)
        at org.jboss.weld.bootstrap.events.AbstractContainerEvent.fire(AbstractContainerEvent.java:53)
        at org.jboss.weld.bootstrap.events.AbstractDeploymentContainerEvent.fire(AbstractDeploymentContainerEvent.java:35)
        ... 71 more
Caused by: java.lang.AbstractMethodError: org/eclipse/microprofile/config/spi/ConfigSource.getPropertyNames()Ljava/util/Set;
        at io.smallrye.config.ConfigValueConfigSourceWrapper.getPropertyNames(ConfigValueConfigSourceWrapper.java:53)
        at io.smallrye.config.SmallRyeConfigSourceInterceptor.iterateNames(SmallRyeConfigSourceInterceptor.java:37)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.SmallRyeConfigSourceInterceptor.iterateNames(SmallRyeConfigSourceInterceptor.java:33)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.SmallRyeConfigSourceInterceptor.iterateNames(SmallRyeConfigSourceInterceptor.java:33)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.SmallRyeConfigSourceInterceptor.iterateNames(SmallRyeConfigSourceInterceptor.java:33)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.SmallRyeConfigSourceInterceptor.iterateNames(SmallRyeConfigSourceInterceptor.java:33)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.SmallRyeConfigSourceInterceptor.iterateNames(SmallRyeConfigSourceInterceptor.java:33)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:58)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:58)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.ProfileConfigSourceInterceptor.iterateNames(ProfileConfigSourceInterceptor.java:96)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:58)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
        at io.smallrye.config.SmallRyeConfig.getPropertyNames(SmallRyeConfig.java:224)
        at io.smallrye.config.inject.ConfigExtension.validate(ConfigExtension.java:129)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
        ... 85 more

```

--- a/roaster-bom/pom.xml
+++ b/roaster-bom/pom.xml
@@ -28,7 +28,7 @@
 
 		<version.org.jboss.weld.weld-junit5>2.0.1.Final</version.org.jboss.weld.weld-junit5>
 		<version.org.jboss.weld.core>3.1.5.Final</version.org.jboss.weld.core>
-		<version.org.jboss.resteasy.client.microprofile>4.5.8.Final</version.org.jboss.resteasy.client.microprofile>
+		<version.org.jboss.resteasy.client.microprofile>4.7.5.Final</version.org.jboss.resteasy.client.microprofile>
 		<version.commons.logging>1.2</version.commons.logging>
 
 		<!-- Selenide -->

